src/migrations/20230119000000-create-groups.js

```diff
@@ -0,0 +1,116 @@
+    const loggedUser = '0';
+    const sessionSig = __filename;
+    const auditDescriptor = 'RUN MIGRATIONS';
+    await queryInterface.sequelize.query(
+      `SELECT
+          set_config('audit.loggedUser', '${loggedUser}', TRUE) as "loggedUser",
+          set_config('audit.transactionId', NULL, TRUE) as "transactionId",
+          set_config('audit.sessionSig', '${sessionSig}', TRUE) as "sessionSig",
+          set_config('audit.auditDescriptor', '${auditDescriptor}', TRUE) as "auditDescriptor";`,
+      { transaction },
+    );
+
+    await queryInterface.createTable('Groups', {
+      id: {
+        type: Sequelize.INTEGER,
+        allowNull: false,
+        primaryKey: true,
+        autoIncrement: true,
+      },
+      name: {
+        type: Sequelize.TEXT,
+        unique: true,
+      },
+      createdAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
+      },
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
+      },
+    }, { transaction });
+
+    await queryInterface.createTable('GroupUsers', {
+      id: {
+        type: Sequelize.INTEGER,
+        allowNull: false,
+        primaryKey: true,
+        autoIncrement: true,
+      },
+      userId: {
+        type: Sequelize.INTEGER,
+        allowNull: false,
+        references: {
+          model: {
+            tableName: 'Users',
+          },
+          key: 'id',
+        },
+      },
+      groupId: {
+        type: Sequelize.INTEGER,
+        allowNull: false,
+        references: {
+          model: {
+            tableName: 'Groups',
+          },
+          key: 'id',
+        },
+      },
+      createdAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
+      },
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
+      },
+    }, { transaction });
+
+    await queryInterface.createTable('GroupGrants', {
+      id: {
+        type: Sequelize.INTEGER,
+        allowNull: false,
+        primaryKey: true,
+        autoIncrement: true,
+      },
+      grantId: {
+        type: Sequelize.INTEGER,
+        allowNull: false,
+        references: {
+          model: {
+            tableName: 'Grants',
+          },
+          key: 'id',
+        },
+      },
+      groupId: {
+        type: Sequelize.INTEGER,
+        allowNull: false,
+        references: {
+          model: {
+            tableName: 'Groups',
+          },
+          key: 'id',
+        },
+      },
+      createdAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
+      },
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
+      },
+    }, { transaction });
+  }),
+  down: async (queryInterface) => queryInterface.sequelize.transaction(async (transaction) => {
+    await queryInterface.dropTable('Groups', { transaction });
+    await queryInterface.dropTable('GroupUsers', { transaction });
+    await queryInterface.dropTable('GroupGrants', { transaction });
+  }),
```

Comment:
Can we make the `down `migration functional also? I looks like we will need to drop some of the ZAL tables.  Right now (after droppgin the join tables first and then `Groups`) and running db:migrate again I am getting an error.

Steps:
- make 'Groups' the last one to drop
- run db:migrate
- run db:migrate:undo
- run db:migrate
- `function "ZALNoUpdateFGroups" already exists`

-  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Reply:
[d85fb68](https://github.com/HHS/Head-Start-TTADP/pull/1304/commits/d85fb68d284bcb23de0811dd3eeed6ace0fb69b4)